x/operator/types/expected_keepers.go

```diff
@@ -0,0 +1,99 @@
+package types
+
+import (
+	sdkmath "cosmossdk.io/math"
+	delegationtype "github.com/ExocoreNetwork/exocore/x/delegation/types"
+	tmprotocrypto "github.com/cometbft/cometbft/proto/tendermint/crypto"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type ExpectDelegationInterface interface {
+	DelegationStateByOperatorAssets(ctx sdk.Context, operatorAddr string, assetsFilter map[string]interface{}) (map[string]map[string]delegationtype.DelegationAmounts, error)
+	IterateDelegationState(ctx sdk.Context, f func(restakerID, assetID, operatorAddr string, state *delegationtype.DelegationAmounts) error) error
+	UpdateDelegationState(ctx sdk.Context, stakerID string, assetID string, delegationAmounts map[string]*delegationtype.DelegationAmounts) (err error)
+
+	UpdateStakerDelegationTotalAmount(ctx sdk.Context, stakerID string, assetID string, opAmount sdkmath.Int) error
+}
+
+type PriceChange struct {
+	OriginalPrice sdkmath.Int
+	NewPrice      sdkmath.Int
+	Decimal       uint8
+}
+type ExpectOracleInterface interface {
+	GetSpecifiedAssetsPrice(ctx sdk.Context, assetsID string) (sdkmath.Int, uint8, error)
+	GetPriceChangeAssets(ctx sdk.Context) (map[string]*PriceChange, error)
+}
+
+type MockOracle struct{}
+
+func (MockOracle) GetSpecifiedAssetsPrice(_ sdk.Context, _ string) (sdkmath.Int, uint8, error) {
+	return sdkmath.NewInt(1), 0, nil
+}
+
+func (MockOracle) GetPriceChangeAssets(_ sdk.Context) (map[string]*PriceChange, error) {
+	// use USDT as the mock asset
+	ret := make(map[string]*PriceChange, 0)
+	usdtAssetID := "0xdac17f958d2ee523a2206206994597c13d831ec7_0x65"
+	ret[usdtAssetID] = &PriceChange{
+		NewPrice:      sdkmath.NewInt(1),
+		OriginalPrice: sdkmath.NewInt(1),
```

Comment:
Is the expectation that the original price as of the last block or as of genesis?

Reply:
The expected price change will be fully controlled by the Oracle module.  The scheduled time for providing the price change might be configured by the parameter of Oracle. The change should be calculated compared to the previous price stored in the Oracle module. This hasn't been determined yet, the price changes state of assets can also be stored and updated in the operator module.

Reply:
Based on the current usage of this function, it appears to expect the price change to be as of each block. The oracle module's implementation may cause some of this to change.

Reply:
Have discussed this with Leon. The Oracle module will define a hook for this, then the operator module needs to implement the hook. The current implementation here definitely needs to be changed after the hook is defined by the  Oracle module.